tests/AbstractTestCase.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Psr\Log\Util\Tests;
```

Comment:
I totally missed the namespace issue, maybe from #1.

This namespace is different from before, and it's below `autoload-dev`, so it's not available to end users like this.

We should move this under `src` and under `autoload`. Also, we should check if we want to also copy the old namespace, but that would conflict with `psr/log` ^1

Reply:
Oh, I see. I would contend, however, that it still belongs under `tests` with all the namespaces like this, except it should move to `autoload`. After all, these are actual tests that we are running on CI.

About conflict with `1.x`, I don't think two packages that share a namespace is necessarily a problem, and it is definitely not prohibited: I do it sometimes, when it is semantically advantageous. However, I put them under `Util` because they are not part of the spec (AFAIK), so it's just utility implementations which as I mentioned before are IMHO not relevant to the specification, just like any other implementation.

Reply:
Util packages have generally used the `Fig\$Lib` namespace.  The `Psr` namespace is for the specs themselves, period.  We should follow that here.

Including an alias file for the old names is fine if people want to make it a bit easier to transition.  (That is, a file that contains a bunch of `class_alias` statements that people can require if desired.)

Reply:
Arguably, this would be a massive BC break, though. Is there consensus for applying this here? As far as I understand, the purpose of this PR is to port everything with max compatibility - with both the original spec and the current PHP code. Although, I don't really understand why: people who use `psr/log` `1.x` will have this in that package, and people who use `2.x` will need e.g. PHP 8.0 and fixing many BC breaks anyway.

Reply:
With the class aliases, the BC would still be there, and we would also avoid the conflict that I was talking about, so I think it would be the better approach.

Reply:
Would it not be considered a BC-break if the file with aliases has to be included separately and optionally? Perhaps, we could `file`-autoload it?

By the way, maybe a philosophical question, but an interesting one to me: is declaration of class aliases code that declares symbols, or code that has an effect?

Reply:
IMHO it's symbols; but I would `file`-autoload it and wrap all `class_alias`es in `if (! class_exists(...))` so to avoid conflicts.